core/server/worker/src/main/java/alluxio/worker/block/BlockLockManager.java

```diff
@@ -178,19 +179,23 @@ private ClientRWLock getBlockLock(long blockId) {
       // Acquire the lock outside the synchronized section because #acquire might need to block.
       // We shouldn't wait indefinitely in acquire because the another lock for this block could be
       // allocated to another thread, in which case we could just use that lock.
-      blockLock = mLockPool.acquire(1, TimeUnit.SECONDS);
-      if (blockLock != null) {
-        synchronized (mSharedMapsLock) {
-          // Check if someone else acquired a block lock for blockId while we were acquiring one.
-          if (mLocks.containsKey(blockId)) {
-            mLockPool.release(blockLock);
-            blockLock = mLocks.get(blockId);
-          } else {
-            mLocks.put(blockId, blockLock);
+      try {
+        blockLock = mLockPool.acquire(1, TimeUnit.SECONDS);
+        if (blockLock != null) {
+          synchronized (mSharedMapsLock) {
+            // Check if someone else acquired a block lock for blockId while we were acquiring one.
+            if (mLocks.containsKey(blockId)) {
+              mLockPool.release(blockLock);
+              blockLock = mLocks.get(blockId);
+            } else {
+              mLocks.put(blockId, blockLock);
+            }
+            blockLock.addReference();
+            return blockLock;
           }
-          blockLock.addReference();
-          return blockLock;
         }
+      } catch (IOException e) {
+        LOG.warn("IOException occurs while acquiring block lock: " + e.getMessage());
```

Comment:
1. Do not use '+'. Do `LOG.warn("IOException occurs while acquiring block lock with error {}." + e.getMessage());`instead.

2. This `IOException` is caused by an `InterrruptedException`. If an `InterruptedException` is caught here, this means that the caller of getBlockInfo really wants to interrupt this call. It should propagate this Exception to the caller. For now, we do not have such a usecase, a RuntimeException is more reasonable.